cmd/charm/charmcmd/cmd.go

```diff
@@ -46,22 +46,54 @@ func New() cmd.Command {
 			DefaultConfig: os.Getenv(osenv.JujuLoggingConfigEnvKey),
 		},
 	})
-	chcmd.Register(&grantCommand{})
-	chcmd.Register(&listCommand{})
-	chcmd.Register(&loginCommand{})
-	chcmd.Register(&publishCommand{})
-	chcmd.Register(&pullCommand{})
-	chcmd.Register(&pushCommand{})
-	chcmd.Register(&pushResourceCommand{})
-	chcmd.Register(&revokeCommand{})
-	chcmd.Register(&setCommand{})
-	chcmd.Register(&showCommand{})
-	chcmd.Register(&termsCommand{})
-	chcmd.Register(&whoamiCommand{})
+	commands := []cmd.Command{
```

Comment:
I wonder if this might be a little easier if it was a map? It might save you a few loops?